ibm/operator_collection_sdk/plugins/modules/oc_zoscloudbroker.py

```diff
@@ -0,0 +1,410 @@
+
+    # determine if modifications were made on target
+    if error is None:
+        result["changed"] = True
+        result["state"] = crd_resource
+    else:
+        result["error"] = True
+        module.fail_json(msg=str(error), **result)
+
+    # successful module execution, exit and pass results
+    module.exit_json(**result)
+
+def validate_module_parameters(module, result):
+    # make a copy of module params  
+    params_copy = {**module.params}
+
+    if params_copy["state"] == "absent":
+        return params_copy
+
+    # set default values if necessary
+    if "configure" not in params_copy["storage"]:
+        params_copy["storage"]["configure"] = False
+    if "enabled" not in params_copy["storage"]:
+        params_copy["storage"]["enabled"] = False
+
+    # validate storage has either configured or enabled or both are false
+    if params_copy["storage"]["configure"] and params_copy["storage"]["enabled"]:
+        module.fail_json(
+            msg=f"Invalid parameters; Storage parameters 'configure' and 'enabled' cannot both be true.",
+            **result)
+
+    # validate required params are present
+    if params_copy["storage"]["configure"]:
+        required_params = [
+            # "volume_access_mode"
+            # "volume_mode",
+            # "storage_size",
+            "storage_class"
+        ]
+        for param in required_params:
+            if param not in params_copy["storage"] \
+            or params_copy["storage"][param] is None:
+                module.fail_json(
+                    msg=f"Missing required argument '{param}'. The following arguments must be suppled when argument storage.configure is True: {", ".join(required_params)}.",
+                    **result)
+    elif params_copy["storage"]["enabled"]:
+            if "persistent_volume_claim" not in params_copy["storage"] \
+            or params_copy["storage"]["persistent_volume_claim"] == "" \
+            or params_copy["storage"]["persistent_volume_claim"] is None:
+                module.fail_json(
+                    msg=f"Missing required argument 'persistent_volume_claim'. persistent_volume_claim must be suppled when argument storage.enabled is True.",
+                    **result)
+
+    # validate label format if they exist
+    if "labels" in params_copy:
+        labels = {}
+        for label in params_copy["labels"]:
+            if re.match("^[^=]*=[^=]*$", label) is None:
+                module.fail_json(
+                    msg=f"Recieved label with misshapen form: '{label}'. Each label in the list should take the form 'key=value'.",
+                    **result)
+            else:
+                [key, value] = label.split("=")
+                labels[key] = value
+        
+        params_copy["labels"] = labels
+
+    return params_copy
+
+def change_zoscloudbroker_state(params, yaml_object) -> tuple[dict|None, Exception|None]:
```

Comment:
@yemi-kelani can u check the oc_zosendpoint PR and see how the ansible runner was imported and used? This will  avoid duplicate code and get rid of this function entirely.